math/n_bonacci.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * @file
+ * @brief Implementation of the
+ * [N-bonacci](http://oeis.org/wiki/N-bonacci_numbers) series
+ *
+ * @details
+ * In general, in N-bonacci sequence,
+ * we generate sum of preceding N numbers from the next term.
+ *
+ * For example, a 3-bonacci sequence is the following:
+ * 0, 0, 1, 1, 2, 4, 7, 13, 24, 44, 81
+ * In this code we take N and M as input where M is the number of terms
+ * to be printed of the N-bonacci series
+ *
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+#include <algorithm>  /// for std::is_equal, std::swap
+#include <cassert>    /// for assert
+#include <iostream>   /// for IO operations
+#include <vector>     /// for std::vector
+
+/**
+ * @namespace math
+ * @brief Mathematical algorithms
+ */
+namespace math {
+/**
+ * @namespace n_bonacci
+ * @brief Functions for the [N-bonacci](http://oeis.org/wiki/N-bonacci_numbers)
+ * implementation
+ */
+namespace n_bonacci {
+/**
+ * @brief Function implements n-bonacci series
+ * @param n
+ * @param m is the number of terms in the N-Bonacci sequence
+ * @returns the n-bonacci sequence as vector array
+ */
+std::vector<int> N_bonacci(int n, unsigned int m) {
+    /**
+     * This function finds N-bonacci series for @param n value and @param m terms
+     */
+
+    std::vector<int> a(m, 0);  // we create an empty array of size m
+
+    int i = 0;
+    a[n - 1] = 1;  /// we initialise the (n-1)th term as 1 which is the sum of preceding N zeros
+    a[n] = 1;  /// similarily the sum of preceding N zeros and the (N+1)th 1 is also 1
+    for (i = n + 1; i < m; i++) {
```

Comment:
Why not simply use
```c++
    for (int i = n + 1; i < m; i++) {
```
instead of initializing a variable at line 46?

Reply:
Right! Done 👍